content_script.js

```diff
@@ -137,38 +140,104 @@ class HTMLSerializer {
    *
    * @param {Element} element The element being processed, which has the src
    *     attribute.
+   * @param {Set<string>} processedAttributes The Set containing all attributes
+   *     already added to |this.html| for the Element.
    * @private
    */
-  processSrcAttribute(element) {
-    if (element.tagName.toLowerCase() != 'iframe') {
-      this.processSrcHole(element);
-    } 
+  processSrcAttribute(element, processedAttributes) {
+    var tag = element.tagName.toLowerCase();
+    switch (tag) {
+      case 'iframe':
+        break; // Do nothing.
+      case 'source':
+        if (!element.parent || element.parent.tagName.toLowerCase() != 'img') {
```

Comment:
The dom is spec'd to always return the tagName in uppercase (https://www.w3.org/TR/DOM-Level-3-Core/core.html#ID-104682815) so I'd recommend refactoring this to rely on uppercase constants instead of toLowerCase'ing the tags.